static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.tsx

```diff
@@ -107,33 +110,62 @@ function SpanSummaryContent(props: ContentProps) {
   const {data: spanHeaderData} = useSpanMetrics(
     {
       search: MutableSearch.fromQueryObject(filters),
-      fields: [
-        'span.description',
-        'avg(span.duration)',
-        'sum(span.self_time)',
-        'count()',
-      ],
+      fields: ['span.description', 'sum(span.self_time)', 'count()'],
+      sorts: [{field: 'sum(span.self_time)', kind: 'desc'}],
     },
     SpanSummaryReferrer.SPAN_SUMMARY_HEADER_DATA
   );
 
-  const description = spanHeaderData[0]?.['span.description'];
-  const timeSpent = spanHeaderData[0]?.['sum(span.self_time)'];
-  const avgDuration = spanHeaderData[0]?.['avg(span.duration)'];
-  const spanCount = spanHeaderData[0]?.['count()'];
+  // Average span duration must be queried for separately, since it could get broken up into multiple groups if used in the first query
+  const {data: avgDurationData} = useSpanMetrics(
+    {
+      search: MutableSearch.fromQueryObject(filters),
+      fields: ['avg(span.duration)'],
+    },
+    SpanSummaryReferrer.SPAN_SUMMARY_HEADER_DATA
+  );
+
+  const parsedData = parseSpanHeaderData(spanHeaderData);
 
   return (
     <Fragment>
       <SpanSummaryControls />
       <SpanSummaryHeader
         spanOp={spanOp}
-        spanDescription={description}
-        avgDuration={avgDuration}
-        timeSpent={timeSpent}
-        spanCount={spanCount}
+        spanDescription={parsedData?.description}
+        avgDuration={avgDurationData[0]?.['avg(span.duration)']}
+        timeSpent={parsedData?.timeSpent}
+        spanCount={parsedData?.spanCount}
       />
       <SpanSummaryCharts />
       <SpanSummaryTable project={project} />
     </Fragment>
   );
 }
+
+function parseSpanHeaderData(data: Partial<SpanMetricsResponse>[]) {
+  if (!data) {
+    return undefined;
+  }
+
+  if (data.length <= 1) {
```

Comment:
nit: `!data` above already protects from the case where `data.length` is `0` so it's not a problem, but in isolation this check looks like a bug in waiting (if `data.length` is `0`, `<= 1` is still true but `data[0]` will error).

_Extremely_ minor.

```suggestion
  if (data.length === 1) {
```

Reply:
Good callout, yeah this is pretty wonky. `!data` resolves to `false` though if `data.length` is `0` (thanks JavaScript) so I'll change the first conditional to `if (!data || data.length === 0)` as well as your suggestion here!